partials/KafkaConfig.java

```diff
@@ -0,0 +1,108 @@
+{% macro kafkaConfig(asyncapi) %}
+{%- set hasSubscribe = false -%}
+{%- set hasPublish = false -%}
+{%- for channelName, channel in asyncapi.channels() -%}
+    {%- if channel.hasPublish() -%}
+        {%- set hasPublish = true -%}
+    {%- endif -%}
+    {%- if channel.hasSubscribe() -%}
+        {%- set hasSubscribe = true -%}
+    {%- endif -%}
+{%- endfor -%}
+package com.asyncapi.infrastructure;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.*;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+{% if hasPublish %}@EnableKafka{% endif %}
+public class Config {
+{%- if hasSubscribe or hasPublish %}
+    @Value("${kafka.bootstrap-servers:localhost:9092}")
+    private String bootstrapServers;
+{% endif %}
+{%- if hasPublish %}
+    @Value("${kafka.subscribe.pool-timeout:3000}")
+    private long poolTimeout;
+
+    @Value("${kafka.subscribe.amount-of-listeners:3}")
+    private Integer amountOfListeners;
+{% endif %}
+{%- if hasSubscribe %}
+    @Bean
+    public KafkaTemplate<Integer, String> kafkaTemplate() {
+        return new KafkaTemplate<>(producerFactory());
+    }
+
+    @Bean
+    public ProducerFactory<Integer, String> producerFactory() {
+        return new DefaultKafkaProducerFactory<>(producerConfigs());
+    }
+
+    @Bean
+    public Map<String, Object> producerConfigs() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
```

Comment:
Why did we use Integer here? Is it just as an example?

Reply:
just example, explanation regarding key and json is value in upper comment